### ISSUE TITLE
Dynamic template data isn't working

### ISSUE THREAD

**Issue by JonahElbaz** [hamano/flask-mail-sendgrid#2]
When testing the input on my template, everything looks good, and i know im sending the right template because i see it in the email. However template data isnt going through

```
from flask import Flask
from flask_mail_sendgrid import MailSendGrid
from flask_mail import Message

app = Flask(__name__)
app.config['MAIL_SENDGRID_API_KEY'] = 'myapi'
mail = MailSendGrid(app)

class EmailProvider:
    @staticmethod
    def sendEmail(email, code):
        msg = Message("Hello",
                      sender="workingSender",
                      recipients=[email])
        msg.template_id = 'validId'
        msg.dynamic_template_data = {"code": code}
        mail.send(msg)

```

Heres my template

```

    <html>
    <head>
    </head>
    <body>
      <table>
        <tr>
            <td style='padding:10px; text-align:center; font-size:15px; font-family:Arial,Helvetica;'>
                <a href='https://www.tec-it.com' title='Barcode Software by TEC-IT'>
                    <img src='https://www.tec-it.com/pics/banner/web/TEC-IT_Banner_120x42.gif' alt='Barcode Software by TEC-IT' border='0' />
                </a>
                <br/>
                <a href='https://www.tec-it.com' title='Barcode Software by TEC-IT'>Barcode Software</a>
            </td>
            <td>
                <img src='https://barcode.tec-it.com/barcode.ashx?data={{code}}&code=Code128&dpi=96&dataseparator=' alt='Barcode Generator TEC-IT'/>
            </td>
        </tr>
        
    </table>
    </body>
    </html>
  
```

**Comment by JonahElbaz** [comment]
After diving into the repo source code i found the issue in these lines

```
template_data = getattr(message, 'template_data', None) 
if template_data:
    personalization.dynamic_template_data = template_data
```

It seems theres an inconsistency in whats being searched for. In order for dynamic_template_data to be added to personalizations template_data must exists - however this variable is not part of your interface.

The following code works for me:

```
app = Flask(__name__)
app.config['MAIL_SENDGRID_API_KEY'] = 'SG.7CVv-ly_XXXX' #should be env var
mail = MailSendGrid(app)

class EmailProvider:
    @staticmethod
    def sendEmail(email, code):
        msg = Message("Hello",
                      sender= send,
                      recipients=[email])
        msg.template_id = 'apiKey'
        msg.template_data = {"code": code}
        msg.dynamic_template_data = {"code": code}
        mail.send(msg)
```

You could either update the interface or update the variable being searched for. Either way it shouldn't damage any current implementations on update.

Other than that - great tool

**Comment by hamano** [comment]
@ConnorD Could you send PR?
Templates are added recently #1, so I have not used it yet.
There are few people affected, so it is no problem to break API compatibility.

**Comment by JonahElbaz** [comment]
Tried to push up a pr but access was denied.
a temporary solution is to do this 
```
     msg.template_data = {"code": code}
        msg.dynamic_template_data = {"code": code}
```
instead of just
```
        msg.dynamic_template_data = {"code": code}
```

**Comment by ConnorD** [comment]
whoops, looks like I must've changed my mind on the naming of the property between when I implemented and updated the Readme.

**Comment by hamano** [comment]
released 0.2
